test/base.py

```diff
@@ -36,18 +38,60 @@
 TEST_TIMEOUT = os.environ.get("TEST_TIMEOUT", 60)
 
 
-class QiskitExperimentsTestCase(QiskitTestCase):
+# If testtools is installed use that as a (mostly) drop in replacement for
+# unittest's TestCase. This will enable the fixtures used for capturing stdout
+# stderr, and pylogging to attach the output to stestr's result stream.
+if qiskit_experiments.warnings.HAS_TESTTOOLS:
+    import testtools
+
+    class BaseTestCase(testtools.TestCase):
+        """Base test class."""
+
+        # testtools maintains their own version of assert functions which mostly
+        # behave as value adds to the std unittest assertion methods. However,
+        # for assertEquals and assertRaises modern unittest has diverged from
+        # the forks in testtools and offer more (or different) options that are
+        # incompatible testtools versions. Just use the stdlib versions so that
+        # our tests work as expected.
+        assertRaises = unittest.TestCase.assertRaises
+        assertEqual = unittest.TestCase.assertEqual
+
+else:
+
+    class BaseTestCase(unittest.TestCase):
+        """Base test class."""
+
+        pass
+
+
+# TODO: copy enforce_subclasses_call decorator?
+class QiskitExperimentsTestCase(BaseTestCase):
     """Qiskit Experiments specific extra functionality for test cases."""
 
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.__setup_called = False
+        self.__teardown_called = False
+
     def setUp(self):
         super().setUp()
         self.useFixture(fixtures.Timeout(TEST_TIMEOUT, gentle=True))
+        if self.__setup_called:
+            raise ValueError(
+                f"In File: {sys.modules[self.__class__.__module__].__file__}\n"
+                "TestCase.setUp was already called. Do not explicitly call "
+                "setUp from your tests. In your own setUp, use super to call "
+                "the base setUp."
+            )
+        self.__setup_called = True
 
     @classmethod
     def setUpClass(cls):
         """Set-up test class."""
         super().setUpClass()
 
+        warnings.filterwarnings("error", category=DeprecationWarning)
```

Comment:
`QiskitTestCase` hard codes many exceptions to this rule. One thing I wanted to see by setting up this PR and testing against terra `main` was if we could pass without that list of exceptions but it would be hard for us to keep with special deprecation warnings in terra that should be allowed, but thankfully our tests pass without any exceptions.